### PR TITLE
Drawing manager Customizations

### PR DESCRIPTION
**Pull request by shubham1g5**
* Abstracts and Exposes `getCurrentPolygon` to get polygon from drawn circles
* Adds a function `resetDrawing` to reset the DrawingManager to it's initial stage.

### CODE REVIEW

File: library/src/main/java/io/ona/kujaku/manager/DrawingManager.java [hunk 1]
```diff
@@ -106,27 +107,29 @@ public void onAnnotationDragFinished(Circle circle) {
...
-            public boolean onMapClick(@NonNull LatLng point) {
-                final PointF pixel = mapboxMap.getProjection().toScreenLocation(point);
-                List<Feature> features = mapboxMap.queryRenderedFeatures(pixel, (Expression) null, CircleManager.ID_GEOJSON_LAYER);
-
-                if (features.size() == 0 && drawingEnabled) {
-                    if (getCurrentKujakuCircle() != null) {
-                        unsetCurrentCircleDraggable();
-                    } else {
-                        drawCircle(point);
+        if(drawOnClick) {
```
Comment (ekigamba) [inline library/src/main/java/io/ona/kujaku/manager/DrawingManager.java:110]: Why did you add this boolean param? It doesn't seem to serve any purpose according to the library module code. Are you by any chance using it any other way?

Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/manager/DrawingManager.java:110]: Could you also format the file, there should be space between `if` and `(`, codacy should have caught this format/style violation

File: library/src/main/java/io/ona/kujaku/manager/DrawingManager.java [hunk 2]
```diff
@@ -278,6 +285,20 @@ private Polygon stopDrawing() {
...
+        // Refresh
+        this.refresh(false);
+
+        return polygon;
+    }
+
+    /**
+     * @return Returns the up to date drawn polygon at any point in time
+     */
+    public Polygon getCurrentPolygon(){
```
Comment (ekigamba) [inline library/src/main/java/io/ona/kujaku/manager/DrawingManager.java:301]: I am curious, what's the use case of exposing this and using it as it is?

File: library/src/main/java/io/ona/kujaku/manager/DrawingManager.java (hunk 1 above)
Comment (shubham1g5) [reply library/src/main/java/io/ona/kujaku/manager/DrawingManager.java:110]: > Why did you add this boolean param? It doesn't seem to serve any purpose according to the library module code. Are you by any chance using it any other way?

Yes, I am adding these circles as user is walking a certain path without user having to click the map. 

> Could you also format the file, there should be space between if and (, codacy should have caught this format/style violation

Sure, My default style settings are bit different, so avoiding to format the whole file.

File: library/src/main/java/io/ona/kujaku/manager/DrawingManager.java (hunk 2 above)
Comment (shubham1g5) [reply library/src/main/java/io/ona/kujaku/manager/DrawingManager.java:301]: @ekigamba I am measuring the boundary area as it's getting drawn and showing it to the user

File: library/src/main/java/io/ona/kujaku/manager/DrawingManager.java (hunk 1 above)
Comment (shubham1g5) [reply library/src/main/java/io/ona/kujaku/manager/DrawingManager.java]: @ekigamba removed this constructor param since I realized I can get the same behaviour by not calling `startDrawing` and instead direcly calling `drawCircle` as user moves the boundary

Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/manager/DrawingManager.java]: Awesome

File: library/src/main/java/io/ona/kujaku/manager/DrawingManager.java (hunk 2 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/manager/DrawingManager.java:294]: Makes sense

**Comment by shubham1g5** [comment]
@ekigamba Look like both comments on this resolved now, Is there anything pending on my end for this PR ?
